crates/python_utilities/src/ast_printer.rs

```diff
@@ -164,6 +175,148 @@ fn emit_body<'a>(body: &'a [Stmt], str_buffer: &mut WritingBuffer<'a>) {
     }
 }
 
+/**
+ * CustomDisplay represents a mutable pretty printing.
+ * WritingBuffer keep track of the indentation state, so we can't just return string.
+ * This allows, e.g. a nested list items to be double-indented.
+ *
+ * defer - When true, it does not push to str_buffer, and just returns the String value.
+ *         This functionality is partially implemented since we need it only to pick up
+ *         the function name.
+ */
+trait CustomDisplay {
+    fn custom_fmt(&self, str_buffer: &mut WritingBuffer, defer: bool) -> String;
+}
+
+impl CustomDisplay for ast::Expr {
+    fn custom_fmt(&self, str_buffer: &mut WritingBuffer, defer: bool) -> String {
+        fn push(str_buffer: &mut WritingBuffer, defer: bool, s: String) -> String {
+            if !defer {
+                str_buffer.push(Cow::Owned(s.clone()));
+            }
+            s
+        }
+        fn push_inline_list(
+            str_buffer: &mut WritingBuffer,
+            defer: bool,
+            list: &Vec<ast::Expr>,
+        ) -> String {
+            str_buffer.push(Cow::Owned("[".to_string()));
+            let mut idx: usize = 0;
+            for elem in list.iter() {
```

Comment:
`for (idx, elem) in list.iter().enumerate() {`

we don't need to re-implement enumerate.

Reply:
Addressed in b8ddcf6079fe659b1cd1c94fd82d5da6e6238c2f